src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeNestedIfStatementsTests.cs

```diff
@@ -0,0 +1,2617 @@
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if (a && b)
+        {
+            [||]if (c || d)
+            {
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if (a && b && (c || d))
+        {
+        }
+    }
+}");
+        }
+
+
+        [Fact]
+        public async Task MergedWithBitwiseOrExpressionNotParenthesized1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if (a | b)
+        {
+            [||]if (c && d)
+            {
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if (a | b && c && d)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedWithBitwiseOrExpressionNotParenthesized2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if (a && b)
+        {
+            [||]if (c | d)
+            {
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if (a && b && c | d)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedWithMixedExpressions1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if (a || b && c)
+        {
+            [||]if (c == d)
+            {
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if ((a || b && c) && c == d)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedWithMixedExpressions2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if (a == b)
+        {
+            [||]if (b && c || d)
+            {
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b, bool c, bool d)
+    {
+        if (a == b && (b && c || d))
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedWithNestedIfInsideWhileLoop()
+        {
+            // Do not consider the while loop to be a simple block (as might be suggested by some language-agnostic helpers).
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+            while (true)
+                [||]if (b)
+                {
+                }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedWithNestedIfInsideBlockInsideUsingStatement()
+        {
+            // Do not consider the using statement to be a simple block (as might be suggested by some language-agnostic helpers).
```

Comment:
would still fix the helper.

Reply:
Responded to this in https://github.com/dotnet/roslyn/pull/30640#discussion_r229462894